AutoCollection/ClientRequests.ts

```diff
@@ -79,16 +80,30 @@ class AutoCollectClientRequests {
 
         let requestParser = new ClientRequestParser(requestOptions, request);
 
-        // Add the source ikey hash to the request headers, if a value was not already provided.
+        // Add the source correlationId to the request headers, if a value was not already provided.
         // The getHeader/setHeader methods aren't available on very old Node versions, and
         // are not included in the v0.10 type declarations currently used. So check if the
         // methods exist before invoking them.
-        if (client.config && client.config.instrumentationKeyHash &&
-            Util.canIncludeCorrelationHeader(client, requestParser.getUrl()) &&
-            request['getHeader'] && request['setHeader'] &&
-            !request['getHeader'](RequestResponseHeaders.sourceInstrumentationKeyHeader)) {
-            request['setHeader'](RequestResponseHeaders.sourceInstrumentationKeyHeader,
-                client.config.instrumentationKeyHash);
+        if (Util.canIncludeCorrelationHeader(client, requestParser.getUrl()) &&
+            request['getHeader'] && request['setHeader']) {
+            if (client.config && client.config.correlationId) {
+                const correlationHeader = request['getHeader'](RequestResponseHeaders.requestContextHeader);
+                if (correlationHeader) {
+                    const components = correlationHeader.split(",");
+                    const key = `${RequestResponseHeaders.requestContextSourceKey}=`;
+                    if (!components.some((value) => value.substring(0,key.length) === key)) {
+                        request['setHeader'](RequestResponseHeaders.requestContextHeader, `${correlationHeader},${RequestResponseHeaders.requestContextSourceKey}=${client.config.correlationId}`);
+                    }
+                } else {
+                    request['setHeader'](RequestResponseHeaders.requestContextHeader, `${RequestResponseHeaders.requestContextSourceKey}=${client.config.correlationId}`);
```

Comment:
It can be if a user explicitly overrides it. By default we set it to `"cid-v1:"` (the equivalent to .NET's `EmptyCorrelationId`)